Mage/src/main/java/mage/abilities/effects/common/DamageTargetEffect.java

```diff
@@ -154,8 +163,18 @@ public String getText(Mode mode) {
         StringBuilder sb = new StringBuilder();
         String message = amount.getMessage();
         sb.append(this.sourceName).append(" deals ");
-        if (message.isEmpty() || !message.equals("1")) {
-            sb.append(amount);
+        if (textPhrasing == ValuePhrasing.LEGACY){
+            if (!message.equals("1")) {
+                sb.append(amount);
+            }
+        } else if (amount instanceof StaticValue) {
+            sb.append(((StaticValue)amount).getValue());
+        } else if (textPhrasing == ValuePhrasing.X_IS || textPhrasing == ValuePhrasing.X_HIDDEN) {
+            sb.append("X");
+        } else if (textPhrasing == ValuePhrasing.EQUAL_TO) {
+            // do nothing
+        } else if (amount instanceof MultipliedValue) {
```

Comment:
I don’t like that style of code:
- it’s full of potential bugs and miss combinations (you combine phrasing param and class type in one switch instead split it);
- it will be miss on new ValuePhrasing changes (no throw exception on unknown value);
- it’s overdeveloped - instead using on two lines of code it required 10+ lines.

Old code:
![IMG_0741](https://github.com/user-attachments/assets/8a54ba3b-5e19-4d1b-bc3f-56c6f621edcb)

New code from PR:
![IMG_0742](https://github.com/user-attachments/assets/7323aa7f-0750-4d47-9603-2aa4afad8e2f)

Must use switch with throw error all the time, not “if else”. Current code smells bad. I don’t look at other effects. But if it use/require same logic then whole thing must be rethink/rework.

P.S. Also look at legacy part - it miss empty check in new code. Is that fine?

Reply:
The intricacies of templating require a lot of parameters to generate text correctly. It's not necessarily "overdeveloped".

I agree that a switch would be cleaner, but it is not obvious to me that would in fact be a workable solution.

Reply:
Also:
1. the old code doesn't correctly generate text in many common scenarios
2. if getText method doesn't correctly generate text, it doesn't impact game logic or stability
3. any discrepancies are easily caught by the verify check

Reply:
For possible ideas: I like a text template style for complex use cases instead multiple if/else (text replacing is slower, but much more easy to edit and support).

// prepare template
template = “some [VALUE] in the middle and [OTHER PARAM]…”;

// prepare params
realValue = xxx;
realParam = yyy;

// gen text
template.replace(“[VALUE]”, realValue);
template.replace(“[OTHER PARAM]”, realParam);

Reply:
I think i see what you're saying here:

1. this needs to throw an exception for unsupported phrasings
2. if-else if chains are inherently less readable and less flexible

I took a shot at changing `DamageTargetEffect` and `DrawCardSourceControllerEffect` to use switch statements here. Right now those are the two effects that are being used as a test bed for this pattern.

What remains are the `instanceof` checks for `MultipliedValue` and `StaticValue` complicating such fragments. The base `DynamicValue` class /could/ be made to own states like being static or having a multiplier, but that would require making it into a class, which we've previously decided against. 

If this is still too spaghetti, i'm open to ideas for how this should work instead.

Reply:
> For possible ideas: I like a text template style for complex use cases instead multiple if/else (text replacing is slower, but much more easy to edit and support).
> 
> // prepare template template = “some [VALUE] in the middle and [OTHER PARAM]…”;
> 
> // prepare params realValue = xxx; realParam = yyy;
> 
> // gen text template.replace(“[VALUE]”, realValue); template.replace(“[OTHER PARAM]”, realParam);

This and xenohedron's comments just showed up for me for some reason--I can get behind this too, but im curious to see if you think the switch statment works first.

Reply:
> but im curious to see if you think the switch statment works first

Yes, new switch is better:

![shot_240904_231701](https://github.com/user-attachments/assets/dfc65ef5-eb4e-4ed9-8c5b-ad9c93fa4544)

Reply:
Agree that the new switch is better. (Ideally you'd have one switch instead of two switches on that enum, but I'm not sure whether or not trying to combine them would be overall cleaner.)

Reply:
That's a solid case for the templating idea from JayDi, now that i think about it.